rubix-common/src/test/java/com/qubole/rubix/common/utils/TestClusterUtil.java

```diff
@@ -69,4 +75,45 @@ public void testGetMasterHostname_noConfFound()
     final String hostname = ClusterUtil.getMasterHostname(conf);
     assertEquals(hostname, TEST_DEFAULT_MASTER_HOSTNAME, "Unexpected hostname!");
   }
+
+  @Test
+  public void testPickLastConfigInRubixSiteXML() throws Exception
+  {
+    ClusterUtil.rubixSiteExists = new AtomicReference<>();
+    CacheConfig.setRubixConfigApplied(conf, false);
+    rubixSiteXmlName = workingDirectory + "/../rubix-common/src/test/resources/rubix-site-duplicate-key.xml";
+    CacheConfig.setRubixSiteLocation(conf, rubixSiteXmlName);
+    Configuration configuration = ClusterUtil.applyRubixSiteConfig(conf);
+    assertEquals(configuration.get("rubix.cache.parallel.warmup"), "false", "Configuration is returning wrong value");
+    assertNull(configuration.get("qubole.team"), "Configuration is returning wrong value");
+  }
+
+  @Test
+  public void testRubixConfigOnReapplying() throws Exception
+  {
+    ClusterUtil.rubixSiteExists = new AtomicReference<>();
+    CacheConfig.setRubixConfigApplied(conf, false);
+    rubixSiteXmlName = workingDirectory + "/../rubix-common/src/test/resources/rubix-site.xml";
+    CacheConfig.setRubixSiteLocation(conf, rubixSiteXmlName);
+    Configuration configuration = ClusterUtil.applyRubixSiteConfig(conf);
+    assertEquals(configuration.get("rubix.cache.parallel.warmup"), "true", "Configuration is returning wrong value");
+    assertEquals(configuration.get("rubix.team.example"), "true", "Configuration is returning wrong value");
+    assertNull(configuration.get("qubole.team"), "Configuration is returning wrong value");
+
+    rubixSiteXmlName = workingDirectory + "/../rubix-common/src/test/resources/rrubix-site-duplicate-key.xml";
+    CacheConfig.setRubixSiteLocation(conf, rubixSiteXmlName);
```

Comment:
If this test is checking reapply then we need to call applyRubixSiteConfig again here.